llvm/tools/llvm-mca/Views/InstructionInfoView.cpp

```diff
@@ -12,15 +12,45 @@
 //===----------------------------------------------------------------------===//
 
 #include "Views/InstructionInfoView.h"
+#include "llvm/ADT/StringExtras.h"
 #include "llvm/Support/FormattedStream.h"
 #include "llvm/Support/JSON.h"
+#include "llvm/Support/WithColor.h"
 
 namespace llvm {
 namespace mca {
 
+void InstructionInfoView::getComment(raw_ostream &OS, const MCInst &MCI) const {
+  StringRef S = MCI.getLoc().getPointer();
+  StringRef InstrStr;
+  size_t Pos = 0, PosCmt = 0;
+
+  // Recognized comments are after assembly instructions on the same line.
+  // It is usefull to add in comment scheduling information from architecture
+  // specification.
+  // '#' comment mark is not supported by llvm-mca
+
+  if (Pos = S.find("\n"); Pos != StringRef::npos) {
+    InstrStr = S.take_front(Pos);
```

Comment:
please move the declaration here, as you're not using `InstrStr` outside the encoding if-statement